src/data/query.ts

```diff
@@ -1,38 +1,41 @@
-import type { QueryResponse } from '../pinecone-generated-ts-fetch';
+import type { QueryResponse as GeneratedQueryResponse } from '../pinecone-generated-ts-fetch';
 import { handleApiError } from '../errors';
 import { buildConfigValidator } from '../validator';
-import { SparseValuesSchema } from './upsert';
+import {
+  RecordSparseValuesSchema,
+  RecordIdSchema,
+  RecordValuesSchema,
+} from './types';
 import { Static, Type } from '@sinclair/typebox';
 import { VectorOperationsProvider } from './vectorOperationsProvider';
 
-const nonemptyString = Type.String({ minLength: 1 });
-
 const shared = {
   topK: Type.Number(),
   includeValues: Type.Optional(Type.Boolean()),
   includeMetadata: Type.Optional(Type.Boolean()),
   filter: Type.Optional(Type.Object({})),
 };
 
-const QueryByVectorId = Type.Object({
+const QueryByRecordId = Type.Object({
   ...shared,
-  id: nonemptyString,
+  id: RecordIdSchema,
   vector: Type.Optional(Type.Never()),
   sparseVector: Type.Optional(Type.Never()),
 });
 
 const QueryByVectorValues = Type.Object({
```

Comment:
**question:** Should this be `QueryByRecordValues`?

Reply:
Yes, good catch.

Reply:
Actually no. In this case, when querying the caller is supplying a list of embedding values (a "vector") they've created using an embedding model. They're not associated with values stored in the index.